src/main/java/net/neoforged/neoforge/event/GrindstoneEvent.java

```diff
@@ -159,5 +174,19 @@ public void setNewBottomItem(ItemStack newBottom) {
         public int getXp() {
             return super.getXp();
         }
+
+        /**
+         * @return access to the grindstone's level and position
+         */
+        public ContainerLevelAccess getContainerAccess() {
+            return this.access;
+        }
+
+        /**
+         * @return the player currently using the grindstone
+         */
+        public Player getPlayer() {
```

Comment:
The constructor parameter, field, getter and the parameter in the hook method need to be marked as nullable due to the deprecated fallback passing in null.